src/versionbits.cpp

```diff
@@ -24,17 +144,29 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex
     int nThreshold = Threshold(params);
     int64_t nTimeStart = BeginTime(params);
     int64_t nTimeTimeout = EndTime(params);
+    int nMinLockedBlocks = MinLockedBlocks(params);
+    int64_t nMinLockedTime = MinLockedTime(params);
+    int64_t nActualLockinTime = 0;
+    int nActualLockinBlock = 0;
+
+    if (nPeriod == 0)
+    {
+        // we cannot do anything further -this deployment is not really defined.
+        return THRESHOLD_DEFINED;
+    }
 
     // A block's state is always the same as that of the first of its period, so it is computed based on a pindexPrev
     // whose height equals a multiple of nPeriod - 1.
     if (pindexPrev != NULL)
     {
+        assert(nPeriod);
         pindexPrev = pindexPrev->GetAncestor(pindexPrev->nHeight - ((pindexPrev->nHeight + 1) % nPeriod));
     }
 
-    // Walk backwards in steps of nPeriod to find a pindexPrev whose information is known
+    // Walk backwards in steps of nPeriod to find a pindexPrev which was DEFINED
```

Comment:
check pindexPrev for NULL, if its null, some garbage was put in the fork definition, like a period larger than the current block height.  Return THRESHOLD_DEFINED?